test/execute_build_test.go

```diff
@@ -357,68 +361,89 @@ func testCreateImage(t *testing.T, when spec.G, it spec.S) {
 		}
 	})
 
-	it("can trigger rebuilds", func() {
+	it("can trigger rebuilds with volume cache", func() {
 		cacheSize := resource.MustParse("1Gi")
 
-		expectedResources := corev1.ResourceRequirements{
-			Limits: corev1.ResourceList{
-				corev1.ResourceMemory: resource.MustParse("1G"),
+		volumeCacheConfig := &buildapi.CacheConfig{
+			Volume: &buildapi.VolumeCache{
+				Request: &cacheSize,
 			},
-			Requests: corev1.ResourceList{
-				corev1.ResourceMemory: resource.MustParse("512M"),
+		}
+
+		generateRebuild(&ctx, t, cfg, clients, volumeCacheConfig, testNamespace, clusterBuilderName, serviceAccountName)
+	})
+
+	it("can trigger rebuilds with registry cache", func() {
+		cacheImageTag := cfg.newImageTag() + "-cache"
+
+		registryCacheConfig := &buildapi.CacheConfig{
+			Registry: &buildapi.RegistryCache{
+				Tag: cacheImageTag,
 			},
 		}
+		generateRebuild(&ctx, t, cfg, clients, registryCacheConfig, testNamespace, clusterBuilderName, serviceAccountName)
+	})
+}
+
+func generateRebuild(ctx *context.Context, t *testing.T, cfg config, clients *clients, cacheConfig *buildapi.CacheConfig, testNamespace, clusterBuilderName, serviceAccountName string) {
+	expectedResources := corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			corev1.ResourceMemory: resource.MustParse("1G"),
+		},
+		Requests: corev1.ResourceList{
+			corev1.ResourceMemory: resource.MustParse("512M"),
+		},
+	}
 
-		imageName := fmt.Sprintf("%s-%s", "test-git-image", "cluster-builder")
+	imageName := fmt.Sprintf("%s-%s", "test-git-image", "cluster-builder")
```

Comment:
Maybe it is worth to use different `imageNames`? Since this `generateRebuild` is called twice (for each cache).